src/fal/telemetry/telemetry.py

```diff
@@ -0,0 +1,325 @@
+
+Modifications are made to ensure that the code works with fal.
+"""
+
+import datetime
+import http.client as httplib
+import warnings
+import posthog
+import pkg_resources
+import yaml
+import os
+from pathlib import Path
+import sys
+import uuid
+from functools import wraps
+from typing import Any, Optional
+
+import platform
+
+
+TELEMETRY_VERSION = "0.0.1"
+DEFAULT_HOME_DIR = "~/.fal"
+CONF_DIR = "stats"
+posthog.project_api_key = "phc_Yf1tsGPPb4POvqVjelT3rPPv2c3FH91zYURyyL30Phy"
+FAL_HOME_DIR = os.getenv("FAL_HOME_DIR")
+
+# Validate the input of type string
+def str_param(item: Any, name: str) -> str:
+    if not isinstance(item, str):
+        raise TypeError(
+            f"TypeError: Variable not supported/wrong type: "
+            f"{item}, {name}, should be a str"
+        )
+    return item
+
+
+def opt_str_param(item: Any, name: str) -> Optional[str]:
+    # Can leverage regular string function
+    if item is not None and not isinstance(item, str):
+        raise TypeError(
+            f"TypeError: Variable not supported/wrong type: "
+            f"{item}, {name}, should be a str"
+        )
+    return item
+
+
+def python_version():
+    py_version = sys.version_info
+    return f"{py_version.major}.{py_version.minor}.{py_version.micro}"
+
+
+def is_online():
+    """Check if host is online"""
+    conn = httplib.HTTPSConnection("www.google.com", timeout=1)
+
+    try:
+        conn.request("HEAD", "/")
+        return True
+    except Exception:
+        return False
+    finally:
+        conn.close()
+
+
+# Will output if the code is within a container
+def is_docker():
+    try:
+        cgroup = Path("/proc/self/cgroup")
+        docker_env = Path("/.dockerenv")
+        return (
+            docker_env.exists()
+            or cgroup.exists()
+            and any("docker" in line for line in cgroup.read_text().splitlines())
+        )
+    except OSError:
+        return False
+
+
+def get_os():
+    """
+    The function will output the client platform
+    """
+    return platform.system()
+
+
+def dbt_installed_version():
+    """Returns: dbt version"""
+    try:
+        return pkg_resources.get_distribution("dbt-core").version
+    except ImportError:
+        return
+
+
+def fal_installed_version():
+    """Returns: fal version"""
+    try:
+        return pkg_resources.get_distribution("fal").version
+    except ImportError:
+        return
+
+
+def is_airflow():
+    """Returns: True for Airflow env"""
+    return "AIRFLOW_CONFIG" in os.environ or "AIRFLOW_HOME" in os.environ
+
+
+def get_home_dir():
+    """
+    Checks if fal home was set through the env variable.
+    returns the actual home_dir path.
+    """
+    return FAL_HOME_DIR if FAL_HOME_DIR else DEFAULT_HOME_DIR
+
+
+def check_dir_exist(input_location=None):
+    """
+    Checks if a specific directory exists, creates if not.
+    In case the user didn't set a custom dir, will turn to the default home
+    """
+    home_dir = get_home_dir()
```

Comment:
What if `home_dir` is None?

Reply:
Not sure if it can be None, since we are defaulting to DEFAULT_HOME_DIR which is hard-coded to ~/.fal. Are you saying FAL_HOME_DIR might be set to something like "None"?